product_operating_unit/models/product_category.py

```diff
@@ -0,0 +1,24 @@
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import api, fields, models
+
+
+class ProductCategory(models.Model):
+    _inherit = 'product.category'
+
+    operating_unit_ids = fields.Many2many(
+        'operating.unit',
+        'product_category_operating_unit_rel',
+        string='Operating Units')
+
+    @api.multi
+    def write(self, vals):
+        if vals.get('operating_unit_ids'):
+            products = self.env['product.template'].search([
+                ('categ_id', 'child_of', self.id)])
+            for product in products:
+                ou_ids = product.operating_unit_ids.ids
+                ou_ids.extend(vals.get('operating_unit_ids')[0][2])
+                product.operating_unit_ids = [(6, 0, ou_ids)]
+        return super().write(vals)
```

Comment:
Why not super(ProductCategory, self).write(vals)?
Why not doing that before the code above (linesn 17-23) so that if it fails we are not changing the products?